hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/HeapNearCacheRecordMap.java

```diff
@@ -96,7 +96,9 @@ public <C extends EvictionCandidate<K, V>> boolean tryEvict(C evictionCandidate,
         if (evictionCandidate == null) {
             return false;
         }
-        if (remove(evictionCandidate.getAccessor()) == null) {
+        V record = remove(evictionCandidate.getAccessor());
```

Comment:
Can we put this check inside `onEvict` and `onExpire` methods? Because normal stats updates are happening inside them and this condition check can also go there to have them in one place.

Additionally we have  a `AbstractNearCacheRecordStore#canUpdateStats` method which is doing desired condition check, it can also be used.